R/RestRserveResponse.R

```diff
@@ -98,40 +101,106 @@ RestRserveResponse = R6::R6Class(
           'text/plain' = as.character,
           identity)
       }
-      if (isTRUE(getOption('RestRserve_RuntimeAsserts', TRUE))) {
-        checkmate::assert_function(serializer, null.ok = TRUE)
-      }
-
       self$serializer = serializer
       self$content_type = content_type
       return(content_type)
     },
     set_status_code = function(code) {
-      checkmate::assert_int(code, lower = 100L, upper = 600L)
+      if (isTRUE(getOption('RestRserve_RuntimeAsserts', TRUE))) {
+        checkmate::assert_int(code, lower = 100L, upper = 600L)
+      }
       self$status_code = code
       return(code)
     },
-    set_header = function(name, value) {
-      checkmate::assert_string(name)
-      checkmate::assert_string(value)
-      self$headers[[name]] = value
-      return(value)
+    has_header = function(name) {
+      if (isTRUE(getOption('RestRserve_RuntimeAsserts', TRUE))) {
+        checkmate::assert_string(name)
+      }
+      return(!is.null(self$headers[[name]]))
     },
     get_header = function(name) {
-      checkmate::assert_string(name)
+      if (isTRUE(getOption('RestRserve_RuntimeAsserts', TRUE))) {
+        checkmate::assert_string(name)
+      }
       return(self$headers[[name]])
     },
+    set_header = function(name, value) {
+      if (isTRUE(getOption('RestRserve_RuntimeAsserts', TRUE))) {
+        checkmate::assert_string(name)
+        checkmate::assert_string(value)
+      }
+      self$headers[[name]] = value
+      return(value)
```

Comment:
it makes sense to consistently invisibly return `value` in all `set` and `append` methods